src/Abuse/Adapters/Redis.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+namespace Utopia\Abuse\Adapters;
+
+use Utopia\Abuse\Adapter;
+use Redis as Client;
+
+class Redis implements Adapter
+{
+    public const NAMESPACE = 'abuse';
+
+    /**
+     * @var Client
+     */
+    protected Client $redis;
+
+    /**
+     * @var string
+     */
+    protected string $key = '';
+
+    /**
+     * @var int
+     */
+    protected int $time;
+
+    /**
+     * @var int
+     */
+    protected int $limit = 0;
+
+    /**
+     * @var int|null
+     */
+    protected ?int $count = null;
+
+    /**
+     * @var array<string, string>
+     */
+    protected array $params = [];
+
+
+    public function __construct(string $key, int $limit, int $seconds, Client $redis)
+    {
+        $this->redis = $redis;
+        $this->key = $key;
+        $time = (int) \date('U', (int) (\floor(\time() / $seconds)) * $seconds); // todo: any good Idea without time()?
```

Comment:
what's the problem?

Reply:
I don't understand? if you are talking about TODO it was already there, not sure if it's relevant